pgjdbc/src/main/java/com/yugabyte/ysql/TopologyAwareLoadBalancer.java

```diff
@@ -113,17 +119,21 @@ public boolean isExplicitFallbackOnly() {
   }
 
   @Override
-  public boolean isHostEligible(Map.Entry<String, LoadBalanceService.NodeInfo> e) {
+  public boolean isHostEligible(Map.Entry<String, LoadBalanceService.NodeInfo> e,
```

Comment:
Please add a detailed comment on the logic here as there are a lot of flags and states here. Will be good from maintenance and code readability pov.